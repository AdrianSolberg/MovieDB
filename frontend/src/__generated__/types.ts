export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
  Date: { input: any; output: any; }
};

export type Movie = {
  _id: Scalars['Int']['output'];
  title: Scalars['String']['output'];
  vote_average: Scalars['Float']['output'];
  vote_count: Scalars['Int']['output'];
  status: Scalars['String']['output'];
  release_date: Scalars['Date']['output'];
  revenue: Scalars['Float']['output'];
  runtime: Scalars['Int']['output'];
  backdrop_path?: Maybe<Scalars['String']['output']>;
  budget: Scalars['Int']['output'];
  homepage?: Maybe<Scalars['String']['output']>;
  imdb_id?: Maybe<Scalars['String']['output']>;
  original_language: Scalars['String']['output'];
  original_title: Scalars['String']['output'];
  overview: Scalars['String']['output'];
  popularity: Scalars['Float']['output'];
  poster_path?: Maybe<Scalars['String']['output']>;
  tagline?: Maybe<Scalars['String']['output']>;
  genres: Array<Scalars['String']['output']>;
  production_companies: Array<Scalars['String']['output']>;
  production_countries: Array<Scalars['String']['output']>;
  spoken_languages: Array<Scalars['String']['output']>;
  keywords: Array<Scalars['String']['output']>;
};

export type Filters = {
  Genre: Array<Scalars['String']['output']>;
  Rating: Array<Scalars['String']['output']>;
  Decade: Array<Scalars['String']['output']>;
  Status: Array<Scalars['String']['output']>;
  Runtime: Array<Scalars['String']['output']>;
};

export type FiltersInput = {
  Genre: Array<Scalars['String']['input']>;
  Rating: Array<Scalars['String']['input']>;
  Decade: Array<Scalars['String']['input']>;
  Status: Array<Scalars['String']['input']>;
  Runtime: Array<Scalars['String']['input']>;
};

export enum SortingType {
  NewestFirst = 'NEWEST_FIRST',
  OldestFirst = 'OLDEST_FIRST',
  BestRated = 'BEST_RATED',
  WorstRated = 'WORST_RATED',
  LongestRuntime = 'LONGEST_RUNTIME',
  ShortestRuntime = 'SHORTEST_RUNTIME'
}

export type Query = {
  movie?: Maybe<Movie>;
  movies: Array<Movie>;
  filters?: Maybe<Filters>;
};


export type QueryMovieArgs = {
  id: Scalars['Int']['input'];
};


export type QueryMoviesArgs = {
  skip?: InputMaybe<Scalars['Int']['input']>;
  limit?: InputMaybe<Scalars['Int']['input']>;
  filters?: InputMaybe<FiltersInput>;
  sortOption?: InputMaybe<SortingType>;
  search?: InputMaybe<Scalars['String']['input']>;
};

export type GetMovieQueryVariables = Exact<{
  movieId: Scalars['Int']['input'];
}>;


export type GetMovieQuery = { __typename?: 'Query', movie?: { __typename?: 'Movie', _id: number, title: string, vote_average: number, vote_count: number, status: string, release_date: any, revenue: number, runtime: number, backdrop_path?: string | null, budget: number, homepage?: string | null, imdb_id?: string | null, original_language: string, original_title: string, overview: string, popularity: number, poster_path?: string | null, tagline?: string | null, genres: Array<string>, production_companies: Array<string>, production_countries: Array<string>, spoken_languages: Array<string>, keywords: Array<string> } | null };

export type GetMoviesQueryVariables = Exact<{
  skip?: InputMaybe<Scalars['Int']['input']>;
  limit?: InputMaybe<Scalars['Int']['input']>;
  filters?: InputMaybe<FiltersInput>;
  sortOption?: InputMaybe<SortingType>;
  search?: InputMaybe<Scalars['String']['input']>;
}>;


export type GetMoviesQuery = { __typename?: 'Query', movies: Array<{ __typename?: 'Movie', _id: number, title: string, vote_average: number, release_date: any, runtime: number, poster_path?: string | null }> };

export type GetFiltersQueryVariables = Exact<{ [key: string]: never; }>;


export type GetFiltersQuery = { __typename?: 'Query', filters?: { __typename?: 'Filters', Genre: Array<string>, Rating: Array<string>, Decade: Array<string>, Status: Array<string>, Runtime: Array<string> } | null };
