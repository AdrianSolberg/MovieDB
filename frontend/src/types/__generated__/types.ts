export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = {
    [K in keyof T]: T[K];
};
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
    [SubKey in K]?: Maybe<T[SubKey]>;
};
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
    [SubKey in K]: Maybe<T[SubKey]>;
};
export type MakeEmpty<
    T extends { [key: string]: unknown },
    K extends keyof T,
> = { [_ in K]?: never };
export type Incremental<T> =
    | T
    | {
          [P in keyof T]?: P extends " $fragmentName" | "__typename"
              ? T[P]
              : never;
      };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
    ID: { input: string; output: string };
    String: { input: string; output: string };
    Boolean: { input: boolean; output: boolean };
    Int: { input: number; output: number };
    Float: { input: number; output: number };
    DateTime: { input: Date; output: Date };
};

export type Movie = {
    __typename?: "Movie";
    _id: Scalars["Int"]["output"];
    title: Scalars["String"]["output"];
    vote_average: Scalars["Float"]["output"];
    vote_count: Scalars["Int"]["output"];
    status: Scalars["String"]["output"];
    release_date: Scalars["DateTime"]["output"];
    revenue: Scalars["Float"]["output"];
    runtime: Scalars["Int"]["output"];
    backdrop_path?: Maybe<Scalars["String"]["output"]>;
    budget: Scalars["Int"]["output"];
    homepage?: Maybe<Scalars["String"]["output"]>;
    imdb_id?: Maybe<Scalars["String"]["output"]>;
    original_language: Scalars["String"]["output"];
    original_title: Scalars["String"]["output"];
    overview: Scalars["String"]["output"];
    popularity: Scalars["Float"]["output"];
    poster_path?: Maybe<Scalars["String"]["output"]>;
    tagline?: Maybe<Scalars["String"]["output"]>;
    genres: Array<Scalars["String"]["output"]>;
    production_companies: Array<Scalars["String"]["output"]>;
    production_countries: Array<Scalars["String"]["output"]>;
    spoken_languages: Array<Scalars["String"]["output"]>;
    keywords: Array<Scalars["String"]["output"]>;
    reviews: Array<Review>;
};

export type Review = {
    __typename?: "Review";
    _id: Scalars["ID"]["output"];
    movie: Movie;
    username: Scalars["String"]["output"];
    rating: Scalars["Int"]["output"];
    comment: Scalars["String"]["output"];
    date: Scalars["DateTime"]["output"];
};

export type Filters = {
    __typename?: "Filters";
    Genre: Array<Scalars["String"]["output"]>;
    Rating: Array<Scalars["String"]["output"]>;
    Decade: Array<Scalars["String"]["output"]>;
    Status: Array<Scalars["String"]["output"]>;
    Runtime: Array<Scalars["String"]["output"]>;
};

export type FiltersInput = {
    Genre: Array<Scalars["String"]["input"]>;
    Rating: Array<Scalars["String"]["input"]>;
    Decade: Array<Scalars["String"]["input"]>;
    Status: Array<Scalars["String"]["input"]>;
    Runtime: Array<Scalars["String"]["input"]>;
};

export enum SortingType {
    MOST_POPULAR = "MOST_POPULAR",
    NEWEST_FIRST = "NEWEST_FIRST",
    OLDEST_FIRST = "OLDEST_FIRST",
    BEST_RATED = "BEST_RATED",
    WORST_RATED = "WORST_RATED",
    LONGEST_RUNTIME = "LONGEST_RUNTIME",
    SHORTEST_RUNTIME = "SHORTEST_RUNTIME",
}

export type Query = {
    __typename?: "Query";
    movie?: Maybe<Movie>;
    movies: Array<Movie>;
    filters: Filters;
    latestReviews: Array<Review>;
    userReviews: Array<Review>;
};

export type QuerymovieArgs = {
    id: Scalars["Int"]["input"];
};

export type QuerymoviesArgs = {
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
    filters?: InputMaybe<FiltersInput>;
    sortOption?: InputMaybe<SortingType>;
    search?: InputMaybe<Scalars["String"]["input"]>;
};

export type QuerylatestReviewsArgs = {
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
};

export type QueryuserReviewsArgs = {
    username: Scalars["String"]["input"];
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
};

export type Mutation = {
    __typename?: "Mutation";
    addReview: Movie;
    deleteReview: Movie;
};

export type MutationaddReviewArgs = {
    movie_id: Scalars["Int"]["input"];
    username: Scalars["String"]["input"];
    rating: Scalars["Int"]["input"];
    comment: Scalars["String"]["input"];
};

export type MutationdeleteReviewArgs = {
    _id: Scalars["ID"]["input"];
};

export type GetMoviesQueryVariables = Exact<{
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
    filters?: InputMaybe<FiltersInput>;
    sortOption?: InputMaybe<SortingType>;
    search?: InputMaybe<Scalars["String"]["input"]>;
}>;

export type GetMoviesQuery = {
    __typename?: "Query";
    movies: Array<{
        __typename?: "Movie";
        _id: number;
        title: string;
        vote_average: number;
        release_date: Date;
        runtime: number;
        poster_path?: string | null;
    }>;
};

export type GetMovieQueryVariables = Exact<{
    movieId: Scalars["Int"]["input"];
}>;

export type GetMovieQuery = {
    __typename?: "Query";
    movie?: {
        __typename?: "Movie";
        _id: number;
        title: string;
        vote_average: number;
        vote_count: number;
        status: string;
        release_date: Date;
        revenue: number;
        runtime: number;
        backdrop_path?: string | null;
        budget: number;
        homepage?: string | null;
        imdb_id?: string | null;
        original_language: string;
        original_title: string;
        overview: string;
        popularity: number;
        poster_path?: string | null;
        tagline?: string | null;
        genres: Array<string>;
        production_companies: Array<string>;
        production_countries: Array<string>;
        spoken_languages: Array<string>;
        keywords: Array<string>;
        reviews: Array<{
            __typename?: "Review";
            _id: string;
            username: string;
            rating: number;
            comment: string;
            date: Date;
        }>;
    } | null;
};

export type GetFiltersQueryVariables = Exact<{ [key: string]: never }>;

export type GetFiltersQuery = {
    __typename?: "Query";
    filters: {
        __typename?: "Filters";
        Genre: Array<string>;
        Rating: Array<string>;
        Decade: Array<string>;
        Status: Array<string>;
        Runtime: Array<string>;
    };
};

export type AddReviewMutationVariables = Exact<{
    movieId: Scalars["Int"]["input"];
    username: Scalars["String"]["input"];
    rating: Scalars["Int"]["input"];
    comment: Scalars["String"]["input"];
}>;

export type AddReviewMutation = {
    __typename?: "Mutation";
    addReview: {
        __typename?: "Movie";
        reviews: Array<{
            __typename?: "Review";
            _id: string;
            username: string;
            rating: number;
            comment: string;
            date: Date;
        }>;
    };
};
