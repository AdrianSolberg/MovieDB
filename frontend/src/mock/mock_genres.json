[
    {
        "id": 0,
        "name": "Drama"
    },
    {
        "id": 1,
        "name": "Romance"
    },
    {
        "id": 2,
        "name": "Western"
    },
    {
        "id": 3,
        "name": "Animation"
    },
    {
        "id": 4,
        "name": "War"
    },
    {
        "id": 5,
        "name": "Science Fiction"
    },
    {
        "id": 6,
        "name": "Comedy"
    },
    {
        "id": 7,
        "name": "Fantasy"
    },
    {
        "id": 8,
        "name": "Adventure"
    },
    {
        "id": 9,
        "name": "Crime"
    },
    {
        "id": 10,
        "name": "Thriller"
    },
    {
        "id": 11,
        "name": "Family"
    },
    {
        "id": 12,
        "name": "Action"
    },
    {
        "id": 13,
        "name": "Mystery"
    }
]
