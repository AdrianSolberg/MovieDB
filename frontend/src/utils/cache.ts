import { makeVar } from "@apollo/client";
import { Filters, SortingType } from "@/types/movieTypes";

export const sortOptionVar = makeVar<SortingType>(
    (sessionStorage.getItem("sort_option") as SortingType) ??
        SortingType.NEWEST_FIRST
);


const storedFilters = sessionStorage.getItem("filters");
const initialFilters: Filters = storedFilters ? JSON.parse(storedFilters) : {
    Genre: [],
    Rating: [],
    Decade: [],
    Status: [],
    Runtime: [],
};
export const filtersVar = makeVar<Filters>(initialFilters);

export const searchVar = makeVar<string>(
    sessionStorage.getItem("search") ?? ""
);

export const usernameVar = makeVar<string>(
    localStorage.getItem("username") ?? "Guest"
);
