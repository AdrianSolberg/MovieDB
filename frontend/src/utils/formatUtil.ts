export const formatDate = (date: Date) => {
    return date.toLocaleDateString("en-US", {
        year: "numeric",
        month: "long",
        day: "numeric",
        hour: "2-digit",
        minute: "2-digit",
    });
};

export const formatNumber = (n: number) => {
    return n.toLocaleString("no-NO");
};

export const formatCurrency = (value: number) => {
    return value.toLocaleString("en-US", {
        style: "currency",
        currency: "USD",
        maximumFractionDigits: 0,
    });
};
