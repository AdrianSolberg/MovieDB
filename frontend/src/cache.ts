import { makeVar } from "@apollo/client";
import { SortingType } from "./utils/searchSortAndFilter";

export const sortOptionVar = makeVar<SortingType>(
    (sessionStorage.getItem("sort_option") as SortingType) ??
        SortingType.NEWEST_FIRST
);

export const filtersVar = makeVar<{ [key: string]: string[] }>(
    JSON.parse(sessionStorage.getItem("filters") ?? "{}")
);

export const searchVar = makeVar<string>(
    sessionStorage.getItem("search") ?? ""
);
