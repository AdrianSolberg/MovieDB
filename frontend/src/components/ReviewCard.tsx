import { Button } from "@/shadcn/components/ui/button";
import { Card, CardContent } from "@/shadcn/components/ui/card";
import Ratings from "@/shadcn/components/ui/rating";
import { usernameVar } from "@/utils/cache";
import { formatDate } from "@/utils/formatUtil";
import { Reference, useMutation, useReactiveVar } from "@apollo/client";
import { Trash2 } from "lucide-react";
import { Link } from "react-router-dom";
import {
    AlertDialog,
    AlertDialogAction,
    AlertDialogCancel,
    AlertDialogContent,
    AlertDialogDescription,
    AlertDialogFooter,
    AlertDialogHeader,
    AlertDialogTitle,
    AlertDialogTrigger,
} from "@/shadcn/components/ui/alert-dialog";
import { DELETE_REVIEW } from "../api/queries";
import { Review } from "@/types/__generated__/types";
import { getImageUrl, ImageType } from "@/utils/imageUrl/imageUrl";
import Loader from "./Loader";
import { toast } from "sonner";

interface ReviewCardProps {
    review: Review;
    showPoster?: boolean;
}

const ReviewCard: React.FC<ReviewCardProps> = ({
    review,
    showPoster = true,
}) => {
    const username = useReactiveVar(usernameVar);

    const [deleteReview, { loading }] = useMutation(DELETE_REVIEW, {
        update(cache, { data }) {
            if (!data?.deleteReview) return;
            const deletedRef: string = `Review:${data.deleteReview._id}`;
            cache.modify({
                id: `Movie:${data.deleteReview.movie._id}`,
                fields: {
                    reviews(existingReviewRefs = []) {
                        return existingReviewRefs.filter(
                            (reviewRef: Reference) =>
                                reviewRef.__ref != deletedRef
                        );
                    },
                },
            });
            cache.modify({
                fields: {
                    latestReviews(existingReviewRefs = []) {
                        return existingReviewRefs.filter(
                            (reviewRef: Reference) =>
                                reviewRef.__ref != deletedRef
                        );
                    },
                    userReviews(existingReviewRefs = [], { storeFieldName }) {
                        if (!storeFieldName.includes(username)) {
                            return existingReviewRefs;
                        }
                        return existingReviewRefs.filter(
                            (reviewRef: Reference) =>
                                reviewRef.__ref != deletedRef
                        );
                    },
                },
            });
            cache.evict({ id: deletedRef });
        },
    });

    const handleDeleteReview = (review: Review) => {
        deleteReview({
            variables: {
                id: review._id,
            },
        })
            .then((response) => {
                if (response.data?.deleteReview) {
                    toast.success("Review has been deleted");
                }
            })
            .catch((error) => {
                toast.error("Failed to delete review", {
                    action: {
                        label: "Retry",
                        onClick: () => handleDeleteReview(review),
                    },
                    description:
                        error instanceof Error
                            ? error.message
                            : "An error occurred",
                });
            });
    };

    return (
        <Card>
            <CardContent className="relative p-4">
                {review.username !== "Guest" &&
                    username === review.username && (
                        <AlertDialog>
                            <AlertDialogTrigger asChild>
                                {loading ? (
                                    <Loader
                                        size="sm"
                                        className="absolute right-4 top-4 m-0 h-8 px-2"
                                    />
                                ) : (
                                    <Button
                                        size="sm"
                                        className="absolute right-4 top-4 h-8 px-2"
                                    >
                                        <Trash2 className="h-4 w-4 sm:hidden" />
                                        <span className="hidden sm:inline">
                                            Delete
                                        </span>
                                    </Button>
                                )}
                            </AlertDialogTrigger>

                            <AlertDialogContent>
                                <AlertDialogHeader>
                                    <AlertDialogTitle>
                                        Are you sure you want to delete this
                                        review?
                                    </AlertDialogTitle>
                                    <AlertDialogDescription>
                                        This action cannot be undone and will
                                        permanently delete the review.
                                    </AlertDialogDescription>
                                </AlertDialogHeader>
                                <AlertDialogFooter>
                                    <AlertDialogCancel>
                                        Cancel
                                    </AlertDialogCancel>
                                    <AlertDialogAction
                                        onClick={() =>
                                            handleDeleteReview(review)
                                        }
                                        disabled={loading}
                                    >
                                        {loading ? "Deleting..." : "Continue"}
                                    </AlertDialogAction>
                                </AlertDialogFooter>
                            </AlertDialogContent>
                        </AlertDialog>
                    )}
                {showPoster && (
                    <Link to={`/movie/${review.movie._id}`}>
                        <img
                            title={review.movie.title}
                            src={getImageUrl(
                                ImageType.POSTER,
                                review.movie.poster_path,
                                "w342"
                            )}
                            className="float-left mb-2 mr-6 h-48 w-32 rounded-lg object-cover"
                            alt={`${review.movie.title} poster`}
                        />
                    </Link>
                )}
                <section className={showPoster ? "min-h-[12rem]" : ""}>
                    <section className="mb-2 pr-16">
                        <h2 className="text-lg font-bold sm:text-xl">
                            {review.username}
                        </h2>
                        <time
                            className="text-xs opacity-60 sm:text-sm"
                            dateTime={review.date.toISOString()}
                        >
                            {formatDate(review.date)}
                        </time>
                    </section>
                    <Ratings
                        value={review.rating}
                        variant="yellow"
                        totalstars={5}
                    />
                    {review.comment && (
                        <p className="mt-2 break-words text-sm sm:text-base">
                            {review.comment}
                        </p>
                    )}
                </section>
            </CardContent>
        </Card>
    );
};

export default ReviewCard;
