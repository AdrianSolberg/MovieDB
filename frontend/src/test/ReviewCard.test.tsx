import { Review } from "@/types/__generated__/types";
import { mock_review } from "./mock/util";
import { usernameVar } from "@/utils/cache";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ReviewCard from "@/components/ReviewCard";
import { MockedProvider } from "@apollo/client/testing";
import { MemoryRouter } from "react-router-dom";
import { formatDate } from "@/utils/formatUtil";

const mockReview: Review = mock_review;

describe("ReviewCard", () => {
    const renderComponent = (review: Review, showPoster = false) => {
        return render(
            <MemoryRouter>
                <MockedProvider mocks={[]} addTypename={false}>
                    <ReviewCard review={review} showPoster={showPoster} />
                </MockedProvider>
            </MemoryRouter>
        );
    };

    afterAll(() => {
        usernameVar("Guest");
    });

    it("displays username, data and comment", async () => {
        renderComponent(mockReview);
        expect(screen.getByText("cosmic_dreamer")).toBeInTheDocument();
        expect(screen.getByText("good movie!")).toBeInTheDocument();
        expect(
            screen.getByText(new RegExp(formatDate(mockReview.date)))
        ).toBeInTheDocument();
    });

    it("does not render poster if showPoster is false", async () => {
        renderComponent(mockReview);
        expect(
            screen.queryByAltText("Interstellar poster")
        ).not.toBeInTheDocument();
    });

    it("renders poster if showPoster is true", async () => {
        renderComponent(mockReview, true);
        expect(
            screen.queryByAltText("Interstellar poster")
        ).toBeInTheDocument();
    });

    it("shows delete button for user's own review", () => {
        usernameVar("cosmic_dreamer");
        renderComponent(mockReview);
        expect(
            screen.getByRole("button", { name: /delete/i })
        ).toBeInTheDocument();
    });

    it("does not show delete button for other users review", () => {
        usernameVar("other_user");
        renderComponent(mockReview);
        expect(
            screen.queryByRole("button", { name: /delete/i })
        ).not.toBeInTheDocument();
    });
});
