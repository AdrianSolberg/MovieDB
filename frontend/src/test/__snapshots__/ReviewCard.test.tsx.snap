// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewCard > matches snapshot for review card with poster 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
  >
    <div
      class="relative p-4"
    >
      <a
        href="/movie/157336"
      >
        <img
          alt="Interstellar poster"
          class="float-left mb-2 mr-6 h-48 w-32 rounded-lg object-cover"
          src="https://image.tmdb.org/t/p/w342/gEU2QniE6E77NI6lCU6MxlNBvIx.jpg"
          title="Interstellar"
        />
      </a>
      <section
        class="min-h-[12rem]"
      >
        <section
          class="mb-2 pr-16"
        >
          <p
            class="text-lg font-bold sm:text-xl"
          >
            test_user
          </p>
          <time
            class="text-xs opacity-60 sm:text-sm"
            datetime="2024-11-13T19:41:09.625Z"
          >
            November 13, 2024 at 08:41 PM
          </time>
        </section>
        <div
          class="flex items-center gap-2"
        >
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star text-yellow-200"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
        </div>
        <p
          class="mt-2 break-words text-sm sm:text-base"
        >
          good movie!
        </p>
      </section>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReviewCard > matches snapshot for review card without poster 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
  >
    <div
      class="relative p-4"
    >
      <section
        class=""
      >
        <section
          class="mb-2 pr-16"
        >
          <p
            class="text-lg font-bold sm:text-xl"
          >
            test_user
          </p>
          <time
            class="text-xs opacity-60 sm:text-sm"
            datetime="2024-11-13T19:41:09.625Z"
          >
            November 13, 2024 at 08:41 PM
          </time>
        </section>
        <div
          class="flex items-center gap-2"
        >
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star text-yellow-200"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
        </div>
        <p
          class="mt-2 break-words text-sm sm:text-base"
        >
          good movie!
        </p>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
